### ISSUE TITLE
InvalidOperationException on Closing Connection

### ISSUE THREAD

**Issue by bigbearzhu** [Code-Sharp/WampSharp#281]
Hi @darkl sorry we got some issue urgent possibly due to changes in 18.9.3 on https://github.com/Code-Sharp/WampSharp/commit/6b3c515b78b28886d6de3afae38e4c9e46559d3b#diff-f35424b41b20cd6ac5271fe06a041b53R140:
```
System.InvalidOperationException
   at System.Threading.Tasks.TaskCompletionSource`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].SetResult(System.__Canon)
   at WampSharp.V2.Client.WampSessionClient`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].RaiseConnectionBroken()
   at WampSharp.V2.Client.WampChannel`1[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].OnConnectionClosed(System.Object, System.EventArgs)
   at WebSocket4Net.WebSocket.FireClosed()
   at WebSocket4Net.WebSocket.OnClosed()
```
Can't find a good pattern to reproduce it at the moment, but it does happen a lot for us.

Just wondering would it be safe to just use TrySetResult?

**Comment by darkl** [comment]
I think it should be ok to replace SetResult with TrySetResult.

Elad

**Comment by darkl** [comment]
Resolved in 18.11.1.
